sdk/core/core-http/lib/httpOperationResponse.ts

```diff
@@ -52,6 +52,11 @@ export interface HttpOperationResponse extends HttpResponse {
    */
   parsedBody?: any;
 
+  /**
+   * The error information in JSON or XML
+   */
+  errorBody?: any;
```

Comment:
I see that `errorBody` is something that we have introduced to `HttpOperationResponse` in this PR.
How do other http requests deal with errors?

cc @bterlson

Reply:
Other http requests reject the operation and throw an error.
Here, the error information gets returned as a response.

Reply:
Why wouldn't we want to reject the operation when an error is encountered?

Reply:
Since the error has a custom structure based on what the service might return, I thought it was the norm for `err` in `reject(err)` to only be of type `Error` or it's derivatives

Overall, we could - the response currently takes a structure of `{ response, error, result }`. This is based on the existing implementation. If we go by this, the error information will then be available on the rejected promise instead of the response. Thoughts?

Reply:
Can you elaborate on what you mean by the custom structure?

I think we should stay consistent with how the package treats errors for other protocols today, which is to reject with them. Depending on what the custom fields are I think we can handle those by creating a new Error type that has the extra fields or add the fields to an error we instantiate.

Reply:
I'm also curious; with the `errorBody`, were to transforming that into an error in the service-bus specific code so that management-plane APIs would resolve or reject based on whether the service returned a success or not?

Reply:
Resolving as this part of the code has been moved out of core-http and is tracked in #4922